tests/Transport/AwsAuthV4Test.php

```diff
@@ -198,4 +198,35 @@ public function testSignsWithEnvironmentalCredentials(): void
             );
         }
     }
+
+    /**
+     * @group unit
+     * @depends testSignsWithProvidedCredentials
+     */
+    public function testStripsTrailingDotInHost(): void
+    {
+        $host = $this->_getHost();
+        $hostWithTrailingDot = $host.'.';
+
+        $config = [
+            'persistent' => false,
+            'transport' => 'AwsAuthV4',
+            'aws_access_key_id' => 'foo',
+            'aws_secret_access_key' => 'bar',
+            'aws_session_token' => 'baz',
+            'aws_region' => 'us-east-1',
+            'host' => $hostWithTrailingDot,
+        ];
+        $client = $this->_getClient($config);
+
+        try {
+            $client->request('_stats');
+        } catch (GuzzleException $e) {
+            $guzzleException = $e->getGuzzleException();
+            /** @var \Psr\Http\Message\RequestInterface $request */
```

Comment:
For the PHPStan error we can ignore it like I did (adding 1 to the count)

https://github.com/ruflin/Elastica/blob/7e4de1eebdcbc5a4731fda61e098aaf40a8d0258/phpstan-baseline.neon#L138-L141

or now I see it I think I would add:

```php
use use GuzzleHttp\Exception\RequestException;

// ...
$request = $guzzleException->getRequest();
$this->assertInstanceOf(RequestException::class, $guzzleException);
$request = $guzzleException->getRequest();
```
with the `assertInstanceOf` call PHPStan understands the code and also we assert that it is the right type, I'll create a PR changing the rest of them to try.